### PR TITLE
Profile generalization: cleanup

### PR DESCRIPTION
**Pull request by citizen428**
## What type of PR is this? (check all applicable)

- [X] Refactor

## Description

This PR removes some temporary/no longer needed code that was used during profile migration. There's some more cleanup work I'd like to do after this (primarily around loading the CSV file in the specs) but I'd rather do that in a separate PR.

## Related Tickets & Documents

#12161

## QA Instructions, Screenshots, Recordings

Again, please test as much as you can. 

* Onboarding flows
* Profile updates
* General usage of the site (article browsing, etc.)

### UI accessibility concerns?

None

## Added/updated tests?

- [X] Yes

## [Forem core team only] How will this change be communicated?

- [X] This change does not need to be communicated, and this is why not: this is primarily a refactoring/cleanup

### CODE REVIEW

File: app/assets/stylesheets/views/comments.scss [hunk 1]
```diff
@@ -273,5 +273,5 @@
 
 .personal-template-button:not(.active),
 .moderator-template-button:not(.active) {
-   @extend .crayons-btn--outlined;
+  @extend .crayons-btn--outlined;
```
Comment (citizen428) [inline app/assets/stylesheets/views/comments.scss:276]: Automatic update from commit hook.

File: app/views/users/_metadata.html.erb [hunk 2]
```diff
@@ -6,7 +6,7 @@
           <%= sanitized_sidebar title.titleize %>
         </strong>
         <p class="crayons-definition__value">
-          <%= sanitized_sidebar value %>
```
Comment (citizen428) [inline app/views/users/_metadata.html.erb:9]: Automatic update from commit hook.

File: babel.config.js [hunk 3]
```diff
@@ -55,13 +55,13 @@ module.exports = function (api) {
         '@babel/plugin-proposal-private-property-in-object',
         {
           loose: true,
-        }
+        },
```
Comment (citizen428) [inline babel.config.js:58]: Automatic update from commit hook.

File: docs/installation/windows.md [hunk 4]
```diff
@@ -290,8 +290,8 @@ There are currently two work-arounds.
 
 ### WSL2 and End to End Tests
 
-In order to run Cypress Test Runner, a few extra setup steps are required to run GUIs
```
Comment (citizen428) [inline docs/installation/windows.md:293]: Automatic update from commit hook.

File: docs/self-hosting.md [hunk 5]
```diff
@@ -4,6 +4,8 @@ title: Self Hosting Forem
 
 ## Hosting your own Forem instance
 
-Visit [the official Forem Self-Host playbook](https://github.com/forem/selfhost) in order to host on DigitalOcean, AWS, Google Cloud, or bare metal.
```
Comment (citizen428) [inline docs/self-hosting.md:7]: Automatic update from commit hook.

File: spec/models/navigation_link_spec.rb [hunk 6]
```diff
@@ -13,7 +13,7 @@
     it "validates the icon" do
       navigation_link.icon = "test.png"
       expect(navigation_link).not_to be_valid
-
```
Comment (citizen428) [inline spec/models/navigation_link_spec.rb:16]: Automatic update from commit hook.

**Comment by citizen428** [comment]
Sorry, I had added some reviewers manually and then marking the PR as ready for review added a whole bunch more, I removed some again.

File: app/assets/images/forem-background.svg [hunk 7]
```diff
@@ -1,5 +1,3 @@
...
...1.266 8.051 9.23 6.474c1.204.844 2.738.574 3.581-.631l5.084-7.06c.843-1.206.571-2.743-.632-3.587l-9.23-6.474-11.266-8.051-9.229-6.474c-1.204-.844-2.739-.573-3.581.632l-5.085 7.06c-.842 1.205-.571 2.742.633 3.586l9.229 6.474zm-11.607-7.992-8.173 11.479-6.629 9.27c-.843 1.204-.571 2.742.632 3.586l7.023 4.925c1.204.845 2.738.574 3.581-.63l6.629-9.27 8.173-11.479 6.629-9.27c.842-1.204.571-2.742-.632-3.586l-7.053-5.096c-1.204-.845-2.739-.574-3.581.631l-6.599 9.44zm92.294-6.057 11.266 8.052 9.229 6.473c1.204.844 2.739.574 3.581-.631l5.054-7.231a2.684 2.684 0 0 0-.462-3.617l-9.229-6.473-11.266-8.052-9.229-6.473c-1.204-.845-2.739-.574-3.581.631l-5.055 7.23c-.842 1.206-.571 2.743.633 3.587l9.059 6.504zM444.91 15.63l-8.174 11.478-6.629 9.27c-.842 1.205-.571 2.742.633 3.586l7.052 5.096c1.204.845 2.739.574 3.581-.63l6.629-9.27 8.174-11.479 6.629-9.27c.842-1.204.571-2.742-.633-3.586l-7.053-5.096c-1.203-.845-2.738-.574-3.58.631l-6.629 9.27zm-20.353 62.053-33.593 5.923c-4.263.751-7.462 2.548-7.392 3.944l1.336 8.57c.24 1.366 3.862 1.96 8.125 1.21l33.592-5.923 33.422-5.892c4.263-.752 7.463-2.549 7.393-3.945L466.104 73c-.241-1.366-3.862-1.96-8.125-1.209l-33.422 5.892z" opacity=".05" fill="#3B49DF"/>
```
[the remark this answers is not in the document]
Comment (rhymes) [reply app/assets/images/forem-background.svg:2]: Yes, it's set to prettify SVGs

File: app/controllers/api/v0/organizations_controller.rb [hunk 8]
```diff
@@ -33,7 +33,7 @@ def users
         num = [per_page, 1000].min
         page = params[:page] || 1
 
-        @users = @organization.users.select(USERS_FOR_SERIALIZATION).page(page).per(num)
+        @users = @organization.users.joins(:profile).select(USERS_FOR_SERIALIZATION).page(page).per(num)
```
Comment (rhymes) [inline app/controllers/api/v0/organizations_controller.rb:36]: no N+1 :D

File: app/controllers/api/v0/users_controller.rb [hunk 9]
```diff
@@ -10,7 +10,7 @@ class UsersController < ApiController
       ].freeze
 
       def show
-        relation = User.select(SHOW_ATTRIBUTES_FOR_SERIALIZATION)
+        relation = User.joins(:profile).select(SHOW_ATTRIBUTES_FOR_SERIALIZATION)
```
Comment (rhymes) [inline app/controllers/api/v0/users_controller.rb:13]: I'm starting to think if we should consider (not in this PR), eager loading profiles using `strict_loading: true` see https://blog.saeloun.com/2020/02/25/rails-strict-loading-mode-to-fix-n-1.html - we would need to understand if the profile is always or very often needed when we load the user object

File: db/migrate/20210309064924_remove_profile_columns_from_user.rb [hunk 10]
```diff
@@ -0,0 +1,33 @@
+class RemoveProfileColumnsFromUser < ActiveRecord::Migration[6.0]
```
Comment (rhymes) [inline db/migrate/20210309064924_remove_profile_columns_from_user.rb:1]: YOLO 🤣

**Review by rhymes — APPROVED** [review]
Great stuff @citizen428!

File: app/controllers/api/v0/users_controller.rb (hunk 9 above)
Comment (citizen428) [reply app/controllers/api/v0/users_controller.rb:13]: Yes, I have a comment to this effect somewhere. Not explicitly about strict loading, but about figuring out the answer to this issue so we can decide on the best strategy :-)

File: db/migrate/20210309064924_remove_profile_columns_from_user.rb (hunk 10 above)
Comment (citizen428) [reply db/migrate/20210309064924_remove_profile_columns_from_user.rb:1]: Haha, luckily we removed all their usages in the previous PR which didn't blow up too many things. Well, at least not things we noticed 🤣

**Review by juliannatetreault — APPROVED** [review]
I took these changes for a spin locally and didn't notice any new bugs or breakage! 🚀

File: db/migrate/20210309064924_remove_profile_columns_from_user.rb (hunk 10 above)
Comment (djuber) [reply db/migrate/20210309064924_remove_profile_columns_from_user.rb:1]: this is the exciting culmination of a long process. I'm so very happy to see all these `alter table, drop column` commands.

File: docs/frontend/preact.md [hunk 11]
```diff
@@ -56,7 +56,7 @@ import { tagPropTypes } from '../../../components/common-prop-types';
 
 const SomeComponentUsingTags = ({ tags = [] }) => (
   <ul>
-    {tags.map(tag => (
+    {tags.map((tag) => (
```
Comment (djuber) [inline docs/frontend/preact.md:59]: was this a linter change?

**Review by djuber — APPROVED** [review]

**Review by maestromac — APPROVED** [review]
tested. LGTM!

File: docs/frontend/preact.md (hunk 11 above)
Comment (citizen428) [reply docs/frontend/preact.md:59]: Yes. There were quite a few of them in this PR, looks like lately some things slipped into `main`.

**Comment by citizen428** [comment]
Note to @maestromac and @Zhao-Andy: I didn't want to merge this when nobody with access to fix potential problems is awake, it's not urgent enough to warrant potentially waking someone up over 😃 I therefore tagged this "merge by any core" and leave it up to one of you to merge either today or early next week if you don't want to push out a change like this on a Friday.

**Pull request merged**